### PR TITLE
Refactors project directory structure, Add home page layout, topnavbar & tailwind css for styling

### PR DESCRIPTION
**Pull request by SatyaRajAwasth1**
- Refactor project directory structure
- Adds basic layout for home page & tailwind for styling 
- Adds sample navigation bar with brand name, menus, search bar and link to alumn by year and department

Part of #3 

Some snapshots of view:

image

image

image

### CODE REVIEW

File: src/components/shared/TopNavigation.astro [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          <nav class="w-64">
+            <ul class="grid grid-cols-2 gap-2">
+              {yearsRange.map((year) => (
+                  <li key={year}>
+                    <a href={`/profiles/${year}`} class="text-gray-700 hover:text-blue-600">{year}</a>
+                  </li>
+              ))}
+            </ul>
+            <h3 class="font-bold mt-4 mb-2">Departments</h3>
+            <ul>
```
Comment (openrijal) [inline src/components/shared/TopNavigation.astro]: Can these be taken from a config file please

File: src/components/shared/TopNavigation.astro [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              <li><a href="/profiles/cmp" class="text-gray-700 hover:text-blue-600">COMP</a></li>
+              <li><a href="/profiles/se" class="text-gray-700 hover:text-blue-600">SE</a></li>
+              <li><a href="/profiles/civil" class="text-gray-700 hover:text-blue-600">CIVIL</a></li>
+            </ul>
+          </nav>
+        </div>
+      </div>
+    </li>
+    <li><a href="/events" class="hover:text-gray-300">Events</a></li>
+    <li><a href="/blog" class="hover:text-gray-300">Blog</a></li>
```
Comment (openrijal) [inline src/components/shared/TopNavigation.astro]: Let's hide the menu items we don't have today.

All the menu items could be a separate config file with title and path

Comment (SatyaRajAwasth1) [reply src/components/shared/TopNavigation.astro]: That sounds pretty cool! Let me try.

File: src/components/shared/TopNavigation.astro (hunk 1 above)
Comment (SatyaRajAwasth1) [reply src/components/shared/TopNavigation.astro]: You mean departments and starting year?

File: src/components/shared/TopNavigation.astro (hunk 2 above)
Comment (SatyaRajAwasth1) [reply src/components/shared/TopNavigation.astro]: > Let's hide the menu items we don't have today.
> 
> All the menu items could be a separate config file with title and path

@openrijal @mukezhz  Daii, worked on it. Can you please have a look and approve ?

**Review by mukezhz — APPROVED** [review]
LGTM

**Review by openrijal — APPROVED** [review]

**Pull request merged**